### ISSUE TITLE
Pylint ignores child members when using contextlib.ExitStack with inheritence

### ISSUE THREAD

**Issue by vmalloc** [PyCQA/pylint#2056]
### Steps to reproduce
Run pylint on the following code:

```python
import contextlib

class Parent(object):

    def __enter__(self):
        return self

    def __exit__(self, *_):
        pass

class Child(Parent):
    def func(self):
        pass


with contextlib.ExitStack() as stack:
    s = stack.enter_context(Child())
    s.func()
```

### Current behavior
Pylint complains on the last line: 
```
E: 18, 4: Instance of 'Parent' has no 'func' member (no-member)
```

### Expected behavior
There should be no errors

### pylint --version output
```
pylint 1.8.4,
astroid 1.6.3
Python 3.6.5 (default, Mar 30 2018, 06:41:53)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

**Comment by PCManticore** [comment]
I might miss something, but where is `Child` used? This code triggers an AttributeError when called.

**Comment by vmalloc** [comment]
I updated it shortly after posting... `Child` is used in the line before the last. Still issues the pylint error, but runs fine.

**Comment by jacobtylerwalls** [comment]
Cannot reproduce on main@7dd8d5158ef4dc7cc2dc732c60b8760894cbb47e
